### ISSUE TITLE
auto evolve up pokemon evolution tree

### ISSUE THREAD

**Issue by xxsteven69xx** [Spegeli/PokemoGoBot-GottaCatchEmAll#321]
I was wondering how likely would it be to build a function that would move up an evolution tree. From what i have observed, it only will evolve to the second stage of an evolution. I'm proposing a feature that works as follows.

bulbasaur %60 perfect and ivysaur 73% perfect in inventory
catch bulbasaur %93 perfect
evolve bulbasuar %93 perf. to ivysaur %93 perf.
transfer ivysaur %73 perf.
save candy till Pokemon with better IV's or enough candy to evolve ivysaur to venusaur.

basically tries to evolve your best IV Pokemon to final evolution.

**Comment by KoooDT** [comment]
I did not so long ago I have wrote how to resolve this issue. But the author of the bot did not want to use it :) So at the moment it will only in my version of the bot.
You can change it by youself in method GetPokemonToEvolve

**Comment by XiCynx** [comment]
This would be very difficult to pull of without it always evolving your Ivysaur to a Venasaur.  You can add Bulbasaur and Ivysaur to the PokemonsToEvolve.ini in the Settings folder and it will always evolve your Bulbasaur to Ivysaur if you have one,

You can also set EvolveOnlyPokemonAboveIV in Usersettings.settings to 90-95 so it will only evolve Pokemon that have above that specific IV rating.

You can then set UseTransferPokemonKeepAboveIV to 90-95 to match that so any Pokemon that is not above this IV percent will be transferred and not evolved.

You can then set TransferPokemonKeepDuplicateAmount to 0 so that way it only keep the BEST Pokemon IV percent that you have and will transfer the other ones.

To me, all the settings to accomplish what you want are there, you just need to set them up correctly to do it.

**Comment by xxsteven69xx** [comment]
Kinda like, but only evolve if on list or...
 If
1st stage IV's < 2nd Stage IV's 
then 
evolve 2nd stage to 3rd
else
evolve 1st stage and
transfer old 2nd stage

**Comment by xxsteven69xx** [comment]
this way the bot always focus on evolving best IV pokemon up the tree automatically.

**Comment by XiCynx** [comment]
So with how you're wording it, you are talking about mass evolving instead of evolving on the fly like it currently does by default?  Right now with it set as the default evolve options, there will never be another stage two to compare it to if it is always evolving what it can as it is available.

**Comment by xxsteven69xx** [comment]
That's the problem, it will never auto evolve from one to two to three without intervention. My idea is especially for dratini to dragonite. Evolve from dratini to dragonite. If you catch a dratini with better IV's than say the dragonair you've been saving then evolve the dratini and trash the one with lesser stats, then continue to save candies for stage 3

**Comment by Shakazan** [comment]
Simple.  
Set UsePokemonToEvolveList to True.
Add Dranti and Dragonair to the PokemonsToEvolve.ini file 
Set EvolveOnlyPokemonAboveIV to true 
Set EvolveOnlyPokemonAboveIVValue to 90.

That way your trash (70%) dragonair will never get evolved and the system will wait for a nice 90%+ dranti/dragonite.

**Comment by xxsteven69xx** [comment]
@Shakazan the way is works now it will never save the 100 candies to evolve dragonair. It will always get 25 candies and evolve dratini every time.

**Comment by Shakazan** [comment]
That is true but only if you keep getting loads of 95%IV dratini. (if
that'd the case then please share co-ords!)
